core/src/main/java/org/apache/brooklyn/core/OsgiLauncher.java

```diff
@@ -0,0 +1,557 @@
+import org.apache.brooklyn.core.mgmt.internal.LocalManagementContext;
+import org.apache.brooklyn.core.mgmt.internal.ManagementContextInternal;
+import org.apache.brooklyn.core.mgmt.persist.BrooklynMementoPersisterToObjectStore;
+import org.apache.brooklyn.core.mgmt.persist.BrooklynPersistenceUtils;
+import org.apache.brooklyn.core.mgmt.persist.PersistMode;
+import org.apache.brooklyn.core.mgmt.persist.PersistenceObjectStore;
+import org.apache.brooklyn.core.mgmt.rebind.PersistenceExceptionHandlerImpl;
+import org.apache.brooklyn.core.mgmt.rebind.RebindManagerImpl;
+import org.apache.brooklyn.core.objs.BrooklynTypes;
+import org.apache.brooklyn.core.server.BrooklynServerConfig;
+import org.apache.brooklyn.core.server.BrooklynServerPaths;
+import org.apache.brooklyn.util.core.ResourceUtils;
+import org.apache.brooklyn.util.exceptions.Exceptions;
+import org.apache.brooklyn.util.exceptions.FatalConfigurationRuntimeException;
+import org.apache.brooklyn.util.exceptions.FatalRuntimeException;
+import org.apache.brooklyn.util.exceptions.RuntimeInterruptedException;
+import org.apache.brooklyn.util.guava.Maybe;
+import org.apache.brooklyn.util.io.FileUtil;
+import org.apache.brooklyn.util.os.Os;
+import org.apache.brooklyn.util.stream.Streams;
+import org.apache.brooklyn.util.text.Strings;
+import org.apache.brooklyn.util.time.Duration;
+import org.apache.brooklyn.util.time.Time;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Initializer for brooklyn-core when running in an OSGi environment.
+ *
+ * Temporarily here; should be totally contained in blueprint beans' init-methods.
+ */
+public class OsgiLauncher {
+
+    private static final Logger log = LoggerFactory.getLogger(OsgiLauncher.class);
+
+    private ManagementContext managementContext;
+
+    private boolean ignoreCatalogErrors = true;
+    private boolean ignorePersistenceErrors = true;
+
+    private HighAvailabilityMode highAvailabilityMode = HighAvailabilityMode.DISABLED;
+
+    private PersistMode persistMode = PersistMode.DISABLED;
+    private String persistenceDir = null;
+    private String persistenceLocation = null;
+    private Duration persistPeriod = Duration.ONE_SECOND;
+
+    public void start() {
+        // Configure launcher
+//        ShutdownHandler shutdownHandler = new AppShutdownHandler();
+        try {
+//            initBrooklynProperties();
+            initCatalog();
+
+            // TODO
+//            launcher.stopWhichAppsOnShutdown(stopWhichAppsOnShutdownMode);
+//            launcher.shutdownHandler(shutdownHandler);
+//
+//            computeAndSetApp(launcher, utils, loader);
+//            customize(launcher);
+//        } catch (FatalConfigurationRuntimeException e) {
+//            throw e;
+//        } catch (Exception e) {
+//            throw new FatalConfigurationRuntimeException("Fatal error configuring Brooklyn launch: " + e.getMessage(), e);
+//        }
+//
+//        try {
```

Comment:
Lots of commented out code in this file. Can you remove the parts which are not needed? For the rest perhaps leave a comment how should be solved or why can't be used directly.